perf: consolidate gcroot registration for cached builds

Mic92 opened the pull request:
Replace individual skipped build scheduling with batch gcroot registration to reduce scheduler overhead for locally cached builds.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Centralized registration of per-user GC roots to preserve skipped (already-built) outputs.
  * Poller now verifies multiple named checks (test, failing-test, skippable-test) and pre-caches a skippable test.

* **Bug Fixes**
  * Local-store outputs for skipped builds are preserved and counted instead of re-running builds.

* **Refactor**
  * Removed separate skipped-build pipeline; scheduler now skips and reports skipped/total counts.

* **Chores**
  * Cleaned up scheduler/permission wiring related to the removed skipped-build flow.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

zowoq commented:
This is very fast, noticeable improvement even on smaller repos.

Not sure how easy this would be to do in buildbot but now that skipped builds aren't counted as builds it would be useful to have a count, e.g. built 1, skipped 9, total 10.

zowoq commented:
```
❯ curl https://buildbot.nix-community.org/nix-outputs/nix-community/infra/master/aarch64-darwin.host-darwin01
/nix/store/x3z060mlzmijmbq25bmlipk6yr1k6dia-darwin-system-25.11.830b3f0                            
```

Seems the outputsPath symlinks aren't being updated, I merged https://github.com/nix-community/infra/commit/0f6b0c4cb8f6254ee181d2121111b1c2b72ddf4e but it still points to the previous commit.

zowoq commented:
Thanks for fixing the outputsPath. 

After updating to 1d9956f4fa844cac49d399c035dddb3c2829e13a and removing `jobReportLimit` build failures seems to cause exceptions https://buildbot.nix-community.org/#/builders/6/builds/2397 and the github status seem wrong, nix-build is still pending and only one build failure.

<img width="1676" height="432" alt="buildbot" src="https://github.com/user-attachments/assets/dea8e8ac-a967-4d01-abcf-f682296d0511" />

Mic92 commented:
Fixed the pendadic issue and added a fallback:

https://github.com/nix-community/buildbot-nix/pull/497/commits/eec9c8b44bd78549d72106da8fa448b97e181e9c#diff-2bee4d68d995f51b4fa170c7b9f3bbe42456d381dab78e7cf2c3d9201b922dfb

zowoq commented:
Thank you.
